app/scripts/metamask-controller.js

```diff
@@ -5164,14 +5160,15 @@ export default class MetamaskController extends EventEmitter {
   }
 
   /**
-   * Gets the permitted accounts for the specified origin. Returns an empty
-   * array if no accounts are permitted.
+   * Gets the sorted permitted accounts for the specified origin. Returns an empty
+   * array if no accounts are permitted or the wallet is locked.
```

Comment:
Can we add a comment explaining why there are cases when we ignore the lock and cases when we don't?

Reply:
https://github.com/MetaMask/metamask-extension/pull/28855/commits/63cb7a2ae284801e30af3957fd74b87c90e0b8d3